### PR TITLE
A few small follow-up updates for View Bookmarks after initial PR

### PR DESCRIPTION
**Pull request by hultonha**
This is a small follow-up PR to address a couple of small issues discovered in the initial View Bookmark PR (see https://github.com/o3de/o3de/pull/7855).

### CODE REVIEW

File: Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp [hunk 1]
```diff
@@ -413,102 +419,94 @@ namespace AzToolsFramework
...
         }
 
-        // if the field is not empty then we have a file linked to the prefab.
+        // record an undo step if a local view bookmark component is added and configured
+        ScopedUndoBatch undoBatch("SetupLocalViewBookmarks");
+        undoBatch.MarkEntityDirty(bookmarkComponent->GetEntityId());
+
+        // initialize default locations to 0. This is a temporary solution to match the 12 locations of the legacy system
+        // once there is a UI for the view bookmarks these lines should be removed
+        auto createDefaultBookmarksFn = [this, &registry]()
```
Comment (moraaar) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp:428]: nit: This is only used in the return statement at the end of the function, moving its declaration at the end just before it's used would be nicer.

File: Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp [hunk 2]
```diff
@@ -266,25 +272,28 @@ namespace AzToolsFramework
                     m_bookmarkfileName = bookmarkComponent->GetLocalBookmarksFileName();
                     using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString;
                     AZStd::string bookmarkKey = ViewBookmarksRegistryPath + m_bookmarkfileName;
-                    auto viewBookmarkSettingsKey = FixedValueString::format("%s",bookmarkKey.c_str());
+                    auto viewBookmarkSettingsKey = FixedValueString::format("%s", bookmarkKey.c_str());
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp:275]: Why not just use bookmarkKey directly?

The SettingsRegistry::Visit function accepts a string_view, both of which an AZStd::fixed_string and AZStd::string are convertible to.

File: Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp [hunk 3]
```diff
@@ -413,102 +419,94 @@ namespace AzToolsFramework
...
+
+        // initialize default locations to 0. This is a temporary solution to match the 12 locations of the legacy system
+        // once there is a UI for the view bookmarks these lines should be removed
+        auto createDefaultBookmarksFn = [this, &registry]()
+        {
+            if (m_localBookmarks.empty())
+            {
+                for (int i = 0; i < DefaultViewBookmarkCount; i++)
+                {
+                    AZStd::string finalPath = "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(i);
```
Comment (moraaar) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp:434]: `finalPath` can be const.

nit: Also, only the last + operation is needed inside the loop, all the others can be cached outside.

File: Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp [hunk 4]
```diff
@@ -413,102 +419,94 @@ namespace AzToolsFramework
...
+                    
+                    if (const bool succeed = registry->SetObject(finalPath, ViewBookmark());
+                        !succeed)
```
Comment (moraaar) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp:437]: nit: if the function already returns a boolean seems like a waste to assign it to a variable to then check that variable.

````
if (!registry->SetObject(finalPath, ViewBookmark()))
````

**Review by moraaar — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp (hunk 2 above)
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp]: Good shout, this is fixed 👍

File: Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp (hunk 1 above)
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp]: I've updated where this logic lives now to make it clearer, this is now updated 👍

File: Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp (hunk 3 above)
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp]: This has now been updated 👍

File: Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp (hunk 4 above)
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp]: Good call, now changed!

File: Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp [hunk 5]
```diff
@@ -386,128 +436,71 @@ namespace AzToolsFramework
...
         case ViewBookmarkType::Standard:
-            finalPath = "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(m_localBookmarkCount++);
+            // note: this function is not used currently and will need to be updated when DefaultViewBookmarkCount is removed
+            setRegPath = BookmarkSetRegPath(m_bookmarkFileName, aznumeric_cast<int>(m_localBookmarks.size()) + 1);
             break;
         case ViewBookmarkType::LastKnownLocation:
-            finalPath = "/" + m_bookmarkfileName + "/LastKnownLocation";
+            setRegPath = "/" + m_bookmarkFileName + "/LastKnownLocation";
             break;
         }
```
Comment (moraaar) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp:485]: This switch could have a default case with a warning.

Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp:485]: Good call, please see this PR for the update https://github.com/o3de/o3de/pull/8434 @moraaar

File: Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp [hunk 6]
```diff
@@ -254,78 +287,99 @@ namespace AzToolsFramework
...
+                            }
+                        }
+
+                        return visitedViewBookmarks;
+                    }
+                }
+                else
+                {
+                    // initialize default locations to 0. This is a temporary solution to match the 12 locations of the legacy system
+                    // once there is a UI for the view bookmarks these lines should be removed
```
[the remark this answers is not in the document]
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp:327]: I've created this GHI roadmap item for improved tracking - https://github.com/o3de/o3de/issues/8435

**Pull request merged**